test/offsetarray.jl

```diff
@@ -786,3 +787,8 @@ end
         @test b[i] == a[r[i]]
     end
 end
+
+@testset "proper patition for non-1-indexed vector" begin
+    @test Iterators.partition(OffsetArray(1:10,10), 5) |> collect == [1:5,6:10] # OffsetVector
```

Comment:
This is not a redundant case:

```suggestion
    @test Iterators.partition(OffsetArray(1:10,10), 5) |> collect == [1:5,6:10] # OffsetVector
    @test Iterators.partition(OffsetArray(collect(1:10),10), 5) |> collect == [1:5,6:10] # OffsetVector
```